crates/rollup/src/driver/context/standalone.rs

```diff
@@ -65,28 +66,79 @@ impl StandaloneContext {
         let client = ReqwestProvider::<Ethereum>::new_http(l1_rpc_url);
         let (new_block_tx, new_block_rx) = mpsc::channel(128);
 
-        let mut new_block_hashes = client.watch_blocks().await?.into_stream();
-        let _handle = tokio::spawn(async move {
-            while let Some(hashes) = new_block_hashes.next().await {
-                for hash in hashes {
-                    match client.get_block_by_hash(hash, BlockTransactionsKind::Full).await {
-                        Ok(Some(block)) => {
-                            let block_with_txs = parse_reth_rpc_block(block);
-                            if let Err(e) = new_block_tx.try_send(block_with_txs) {
-                                error!("Failed to send new block to channel: {:?}", e);
+        let _handle = match client.watch_blocks().await {
+            Ok(new_block_hashes) => tokio::spawn(async move {
+                let mut stream = new_block_hashes.into_stream();
+                while let Some(hashes) = stream.next().await {
+                    for hash in hashes {
+                        match client.get_block_by_hash(hash, true.into()).await {
+                            Ok(Some(block)) => {
+                                let block_with_txs = parse_reth_rpc_block(block);
+                                if let Err(e) = new_block_tx.try_send(block_with_txs) {
+                                    error!("Failed to send new block to channel: {:?}", e);
+                                }
+                            }
+                            Ok(None) => {
+                                // Q: does `eth_getBlockByHash` return a block if it was
+                                // already reorged out? No-op for now.
+                                error!("Failed to get block by hash: block not found");
+                            }
+                            Err(e) => {
+                                error!("Failed to get block by hash: {:?}", e);
                             }
-                        }
-                        Ok(None) => {
-                            // Question: does `eth_getBlockByHash` return a block if it was
-                            // already reorged out? No-op for now.
-                        }
-                        Err(e) => {
-                            error!("Failed to get block by hash: {:?}", e);
                         }
                     }
                 }
+            }),
+            Err(err) => {
+                if err
+                    .as_error_resp()
+                    .is_some_and(|resp| !resp.message.to_lowercase().contains("not supported"))
+                {
+                    // On any error other than "not supported", return the error and fail.
+                    error!("Failed to watch new blocks via HTTP");
+                    return Err(err);
+                }
```

Comment:
I think this is fine, but we should also then allow the user to pass an `--http` flag that uses http polling explicitly instead of subscriptions in order to not get caught in this block.

Reply:
I don't fully get what the flag would mean, could you elaborate? 
If the URL scheme is HTTP we are forced to poll